components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/deployment/AppCreator.java

```diff
@@ -270,7 +271,14 @@ private void addI18nResources(Map<String, Properties> componentI18nResources, I1
         if ((componentI18nResources == null) || componentI18nResources.isEmpty()) {
             return;
         }
-        componentI18nResources.forEach(i18nResources::addI18nResource);
+        componentI18nResources.forEach((localString, properties) -> {
+            Locale locale = Locale.forLanguageTag(localString.replace("_", "-"));
+            if (!locale.getLanguage().isEmpty()) {
+                i18nResources.addI18nResource(locale, properties);
+            } else {
+                throw new UUFException("Cannot deploy language file for " + localString);
```

Comment:
Shall we add a more detailed error message here. Clearly we NOT deploying language files.